app/Models/Beatmap.php

```diff
@@ -279,6 +289,34 @@ public function getAttribute($key)
         };
     }
 
+    /**
+     * @return Collection<User>
+     */
+    public function getOwners(): Collection
+    {
+        $owners = $this->beatmapOwners->loadMissing('user')->map(
+            fn ($beatmapOwner) => $beatmapOwner->user ?? new DeletedUser(['user_id' => $beatmapOwner->user_id])
+        );
+
+        // TODO: remove when everything writes to beatmap_owners.
```

Comment:
more like should eventually decide whether an empty collection is acceptable or not (in case of broken owners or just simplifying simple cases)